### PR TITLE
Disable xcode indexing in CI via COMPILER_INDEX_STORE_ENABLE=NO argument

### PR DESCRIPTION
**Pull request by andrijasinski**
## Description

Pull request in hand is adding additional logic to disable project indexing with `xcodebuild` command if the environment variable `CI` exists and equals to `true`. 

## Related Issues

* #37231

## Tests

No tests added. 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jmagman** [comment]
I don't think we should hard-code that "CI" environment variable.

There's zero Xcode documentation on this feature but I see some upstream-clang doc proposal here: https://docs.google.com/document/d/1cH2sTpgSnJZCkZtJl1aY-rzy4uGPcrI-6RrUpdATO2Q:

`support a range of IDE features like jump-to-definition, find usages, and global rename`

Is there any reason for `flutter build ios` to ever enable indexing while building?  Seems like you wouldn't want an index until you actually open Xcode and need those IDE features.

Firebase got a 8% speed up by using COMPILER_INDEX_STORE_ENABLE=NO https://github.com/firebase/firebase-ios-sdk/pull/2452

@cbracken @dnfield What do you think?

**Comment by jonahwilliams** [comment]
I don't think we should expose this through a magic environment variable. If we don't need indexing at all then why not turn it off completely?

this location int the code is only called when we build/run through flutter and not xcode

**Comment by jmagman** [comment]
> I don't think we should expose this through a magic environment variable. If we don't need indexing at all then why not turn it off completely?
> 
> this location int the code is only called when we build/run through flutter and not xcode

That's what I just suggested. There's no reason to have indexing at all from any flutter build command. Seems like you wouldn't want an index until you actually open Xcode and need those IDE features.

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -378,6 +379,10 @@ Future<XcodeBuildResult> buildXcodeProject({
   // e.g. `flutter build bundle`.
   buildCommands.add('FLUTTER_SUPPRESS_ANALYTICS=true');
 
+  if (io.Platform.environment.containsKey('CI') && io.Platform.environment['CI'] == 'true') {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/mac.dart]: Remove this check, we would like to have indexing off for all flutter builds.

**Review by jmagman — CHANGES_REQUESTED** [review]

**Comment by christopherfujino** [comment]
@andrijasinski would you still like to land this (if so, please respond to jmagman's requested changes). If not, I will close and open a new PR.

**Comment by andrijasinski** [comment]
@christopherfujino @jmagman PR is updated to disable indexing for all flutter builds.

**Review by andrijasinski — COMMENTED** [review]
Add `COMPILER_INDEX_STORE_ENABLE=NO` by default

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**